### DISCUSSION TITLE
Searching nodes in a Map: How to know number of found results?

### DISCUSSION THREAD

**Discussion by ldmpub** [freeplane/freeplane#905] [Help]
Hi,

When performing a search in Freeplane, is there a way to **know the number of results returned by the search?** like for instance when searching in a page within Google Chrome (see screenshot below):
![image](https://user-images.githubusercontent.com/1452146/209438086-e06eb03b-4ade-41f5-ae70-5e0bb882f181.png)

The number of results will help user refine his query if there are too many results for his initial search ... and thus gaining time instead of "navigating" among a huge amount of results during several minutes ...

I have searched Freeplane Documentation, Helps forum and Google ... the only solution seems to install an add-on 🤨 => 
https://github.com/lilive/Freeplane-Jumper

**Comment by euu2021** [comment]
`Edit->Find->Find and replace…` may be an option.

**Comment by ldmpub** [comment]
@euu2021 thanks for your suggestion, I've already used the 3 differents searches available in FP (2 in `Edit menu` + 1 via `Toolbar`) but the number of results found is not available in any of these screens 😥.

My suggestion would be to display this info for instance here in the "simple search":
![image](https://user-images.githubusercontent.com/1452146/209469818-0d69293a-67df-42ef-bb75-4a38fe9299d0.png)

The same info would be useful to display in the "Search Toolbar"...

What do you think, **should I create a new issue for this feature request?**

**Reply by EdoFro** [reply]
it may be a "Count" button, that when pressed, it updates the total value in the toolbar or in the status line.

For example in Notepad ++:
![image](https://user-images.githubusercontent.com/35700575/209569993-a12d610b-0645-497e-9694-e2caffd26e99.png)

**Reply by ldmpub** [reply]
Exactly ... does it mean I need to fill in an issue to improve freeplane search feature ?

**Comment by dpolivaev** [comment]
I have suddenly been feeling pretty tired so I need some rest before I come back to make any further improvements. I hope it takes a few days but actually it might take me longer. Therefore don't wonder if I do not respond to the majority of the posts.

But I would like to comment on the discussion.

The best way to convince me to give any feature request high priority is to make me see why it is important. Starting a discussion thread is probably better because here I can see how many people relate to the idea, but if people start commenting on the issues it is fine too.

In any case, I need to understand what you need and why.

Saying that you want to have a feature because some other software needs it is almost never enough unless I think that I understand the reasons and the needs correctly. Here it is not the case. Saying that some other software X has similar features does not explain anything.  And without understanding the needs entirely I can neither find a good solution nor see why you need it.

Because in most places Freeplane checks the conditions lazily and can not count the matches if the feature becomes implemented the count could be added only to the "Find and replace" dialog.  But saying that I still want to admit that I do not see the use cases where it is so overwhelmingly useful.

**Reply by euu2021** [reply]
Me: make a song to motivate Dimitry to work on new features for the FreePlane community

AI: 
Dimitry, oh Dimitry
A programmer of great renown
Your work on FreePlane, it sets you apart
It's time to take your skills to the next level

Chorus:
Come on Dimitry, let's get to work
The FreePlane community needs your help
New features to develop, bugs to fix
We know you're up for the task, Dimitry

Verse 2:
You've got the talent, you've got the drive
To make FreePlane even better than before
So don't let doubt or fear hold you back
Step up and rise to the challenge, Dimitry

Chorus:
Come on Dimitry, let's get to work
The FreePlane community needs your help
New features to develop, bugs to fix
We know you're up for the task, Dimitry

Bridge:
You're not alone, we're all here to support you
Every step of the way, we'll stand beside you
So don't give up, keep pushing forward
We believe in you, Dimitry

Chorus:
Come on Dimitry, let's get to work
The FreePlane community needs your help
New features to develop, bugs to fix
We know you're up for the task, Dimitry

**Reply by euu2021** [reply]
But, seriously, resting is important. Take care of your health.

**Comment by ldmpub** [comment]
Follow-up: I have installed and I'm currently using [Jumper](https://github.com/lilive/Freeplane-Jumper/releases) to answer this need.
Note: it's the first freeplane add-on I have ever installed 😁.

To answer @dpolivaev request to clarify the use case, I reproduce below my initial post:
> The number of results will help user refine his query if there are too many results for his initial search ... and thus gaining time instead of "navigating" among a huge amount of results during several minutes ...

Note that I will put this request in stand by because I'm aware that currently there are big great features on the workbench and particularly the "New nodes Layouts" 💪.

**Reply by bepolymathe** [reply]
Hi, 

Never let go... but know how to rest 😸 

> Follow-up: I have installed and I'm currently using [Jumper](https://github.com/lilive/Freeplane-Jumper/releases) to answer this need.

Yes, but the problem is that this addon is no longer maintained and is no longer working for [me](https://github.com/lilive/Freeplane-Jumper/issues/8).

On the usage side, I would eventually have the use to count todo when I search by the "todo" style... but I get around this problem with a formula in this type of central node details...
 
```
=int count(styleName) {
    node.find { it.style.name == styleName }.size()
}
def t = 'Tâches'
def na = 'Next'
def w = 'Waiting'
def d = 'Done'
"""
| Type | Nombre |
|-|-:|
| Tâches | ${count(t)} |
| Next | ${count(na)} |
| Waiting | ${count(w)} |
| Done | ${count(d)} |
"""
```
![image](https://user-images.githubusercontent.com/29939805/212755361-9404a6d4-e4c1-4b6b-b76b-533b45056034.png)


But this solution is not practical for frequent research on differents terms. That’s why I miss Jumper.

**Reply by ldmpub** [reply]
Oups ... that's one drawback of addons and extensions ... it's a "personal initiative" and thus the risk that it's not maintained 😥 is higher than when it's part of the app itself.
Feel free to add your inputs to improve "native search experience" in Freeplane.

**Reply by bepolymathe** [reply]
To answer the request of @ldmpub 

My node search needs in freeplane are important and are globally covered by the current features. The only thing I miss is autocompletion for quick searches which Jumper used to do very well. In general, it seems to me that the search features are very powerful. It's just sometimes the number of clicks to do that produces a bit of friction when you're in a meeting and looking for a quick info. But I can live with that.

**Reply by euu2021** [reply]
@bepolymathe about Jumper not working for you, did you try the solution mentioned at https://github.com/lilive/Freeplane-Jumper/issues/8#issuecomment-1382904131 ?

**Reply by EdoFro** [reply]
> Oups ... that's one drawback of addons and extensions ... it's a "personal initiative" and thus the risk that it's not maintained 😥 is higher than when it's part of the app itself. Feel free to add your inputs to improve "native search experience" in Freeplane.

In this case the Jumper add-on is open source and anybody can forge it and continue its developement.

Adding more features to Freeplane itself has its risks as well. The central application gets bigger and more complex and difficult to mantain. That's no problem when you have a big team of developers, but this is not the case here.

I prefer that an add-on gets old and not maintained and that Freeplane itself rocks in speed and performance, than trying to add a lot of functionalities to it and that the "personal initiative" of its developer get burned.

jm2c

**Reply by EdoFro** [reply]
In other words. I think that we must try to add most new functionalities as new add-ons or plugins and ease the workload on Dimitry.

**Reply by bepolymathe** [reply]
> @bepolymathe about Jumper not working for you, did you try the solution mentioned at [lilive/Freeplane-Jumper#8 (comment)](https://github.com/lilive/Freeplane-Jumper/issues/8#issuecomment-1382904131) ?

Oh, i didn’t notice that ! Thanks it’s works as a workaround.
